pkg/stub/handler.go

```diff
@@ -252,6 +248,18 @@ func upsertDaemonSet(oa *v1alpha1.OneAgent) error {
 	return nil
 }
 
+// hasSpecChanged compares essential OneAgent custom resource settings with the
+// actual settings in the DaemonSet object
+func hasSpecChanged(dsSpec *appsv1.DaemonSetSpec, cr *v1alpha1.OneAgent) bool {
```

Comment:
This guy has high potential for _unit testing_ :)

Further more putting into a util class could be also considered...